libgnucash/app-utils/test/test-gnc-optiondb.scm

```diff
@@ -0,0 +1,237 @@
+ ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+ ; test-gnc-option.scm -- unit tests for GncOption class.           ;
+ ; Copyright (C) 2019 John Ralls <jralls@ceridwen.us>               ;
+ ;                                                                  ;
+ ; This program is free software; you can redistribute it and/or    ;
+ ; modify it under the terms of the GNU General Public License as   ;
+ ; published by the Free Software Foundation; either version 2 of   ;
+ ; the License, or (at your option) any later version.              ;
+ ;                                                                  ;
+ ; This program is distributed in the hope that it will be useful,  ;
+ ; but WITHOUT ANY WARRANTY; without even the implied warranty of   ;
+ ; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    ;
+ ; GNU General Public License for more details.                     ;
+ ;                                                                  ;
+ ; You should have received a copy of the GNU General Public License;
+ ; along with this program; if not, contact:                        ;
+ ;                                                                  ;
+ ; Free Software Foundation           Voice:  +1-617-542-5942       ;
+ ; 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       ;
+ ; Boston, MA  02110-1301,  USA       gnu@gnu.org                   ;
+ ;                                                                  ;
+ ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(use-modules (srfi srfi-64))
+(use-modules (tests srfi64-extras))
+
+;; This is a special case where we can't use the exported registration function
+;; because we need to transform the default argument first depending on its
+;; Scheme type.
+(define (gnc:register-multichoice-option options section name key docstring default multichoice)
+  (issue-deprecation-warning "gnc:make-multichoice-option is deprecated. Make and register the option in one command with gnc-register-multichoice-option.")
+  (let ((defval (cond ((symbol? default)
+                       (symbol->string default))
+                      ((number? default)
+                       (number->string default))
+                     (else default))))
+  (gnc-register-multichoice-option options section name key docstring defval multichoice)))
```

Comment:
This is slightly contradictory: the comment says this is a special case and we can't use the swigified C++ function, yet we do unconditionally issue a deprecation warning telling users to do exactly that.
Could this value conversion be done in the swig wrapper directly by writing a custom typemap for it?

Reply:
Ok, taking a step back... This is in our unit tests. Does this mean this is only a special case during testing ? In that case we don't have to do anything in our swig wrapper obviously.

Reply:
One more step back I see this function is also in options.scm for the same reason. So the initial suggestions stand (but for options.scm now).
1. do we need the deprecation warning if we can't drop the function or
2. can we work around this in the swig wrapper by playing with typemaps or redefining that particular function ?

Reply:
Unfortunately it's also in options.scm. There are a couple of cases like this caused by Scheme's lax type system. In  this instance some of the multichoice options use symbols and others use numbers. Several calls to `gnc:make-foo-option` pass lambdas that need to be executed to extract the value (which is almost always `'()`. IMO the fix is to change the calling scheme code to remove the need to filter the arguments, but in the meantime the tests need to make sure that the existing calls work.

Reply:
Agreed. That can be a followup task after the merge though.

Reply:
Remember that all of those `gnc:make-foo-option`s are deprecated. I'll clear them out of our code after merge but they'll have to stay in GnuCash 5.x to avoid breaking user code.